app/controllers/api/v1/jwt_api/response_controller.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    module JwtApi
+      class ResponseController < JwtApiController
+        before_action :set_person, only: %i[show index completed]
+        before_action :set_response, only: %i[show create]
+        before_action :set_responses, only: %i[index]
+        before_action :set_completed_responses, only: %i[completed]
+        before_action :check_empty_response, only: %i[create]
+
+        def show
+          render json: @response, serializer: Api::PersonalizedQuestionnaireSerializer
+        end
+
+        def index
+          render json: @responses, each_serializer: Api::ResponseSerializer
+        end
+
+        def create
+          content = ResponseContent.create!(content: response_content)
+          @response.update!(content: content.id)
+          @response.complete!
+          head 201
+        end
+
+        def completed
+          render json: @responses, each_serializer: Api::ResponseSerializer
+        end
+
+        private
+
+        def set_person
+          # Debugging
+          # token = request.headers['Authorization'].split.last
+          # Rails.logger.warn Knock::AuthToken.new(token: token)
+          # Rails.logger.warn current_auth_user.person.inspect
+
+          @person = current_auth_user&.person
+          return if @person.present?
+
+          result = { result: current_auth_user.to_json }
+          render(status: :not_found, json: result)
+        end
+
+        def set_completed_responses
+          @responses = @person.my_completed_responses
+          return if @responses.present?
+
+          result = { result: 'Geen completed responses voor deze persoon gevonden' }
+          render(status: :not_found, json: result)
+        end
```

Comment:
I actually take issue with the fact that if there are no completed responses, that we throw a 404, instead of just returning an empty array with a 200. What are your thoughts on this? Same with the 'open responses' method below.

Because it's not like the user supplied an ID and we couldn't find an object with that ID. It's just that the user asked hey give me all my open responses, and we went sure, here you go: [] (empty array). I don't think it should give a 404 because it's just regular interaction.

I'm running into this problem with the ikia backend API that suddenly gives a 404 when I ask if there are questionnaires to be filled out and there are no new questionnaires to be filled out, which I find weird.

Reply:
good point, ive changed all of them to a 200